File: sdk/julia/src/contacts.jl
# Contacts: manage account contacts
struct ContactsResource
    http::HttpClient
end

# GET /api/contact - Return a list of contacts on this account
function list(r::ContactsResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/contact"; params = params)
end

# POST /api/contact - Create new contact account
function create(r::ContactsResource; kwargs...)
    params = Dict(kwargs)
    http_post(r.http, "/api/contact"; params = params)
end

# GET /api/contact/privileges - List possible contact privileges
function get_privileges(r::ContactsResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/contact/privileges"; params = params)
end

# GET /api/contact/{id} - Get contact details
function get(r::ContactsResource; kwargs...)
    params = Dict(kwargs)
    http_get(r.http, "/api/contact/{id}"; params = params)
end

# PUT /api/contact/{id} - Update contact details
function update(r::ContactsResource; kwargs...)
    params = Dict(kwargs)
    http_put(r.http, "/api/contact/{id}"; params = params)
end

File: sdk/dart/lib/src/contacts.dart
import 'http_client.dart';

/// Contacts: manage account contacts
class ContactsResource {
  final HttpClient httpClient;
  ContactsResource(this.httpClient);

  /// GET /api/contact - Return a list of contacts on this account
  Future<Map<String, dynamic>> list([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/contact', params);
  }

  /// POST /api/contact - Create new contact account
  Future<Map<String, dynamic>> create([Map<String, dynamic>? params]) async {
    return httpClient.post('/api/contact', params);
  }

  /// GET /api/contact/privileges - List possible contact privileges
  Future<Map<String, dynamic>> getPrivileges([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/contact/privileges', params);
  }

  /// GET /api/contact/{id} - Get contact details
  Future<Map<String, dynamic>> get([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/contact/{id}', params);
  }

  /// PUT /api/contact/{id} - Update contact details
  Future<Map<String, dynamic>> update([Map<String, dynamic>? params]) async {
    return httpClient.put('/api/contact/{id}', params);
  }

}
